teuthology/repo_utils.py

```diff
@@ -97,20 +97,25 @@ def enforce_repo_state(repo_url, dest_path, branch, remove_on_error=True):
         raise
 
 
-def clone_repo(repo_url, dest_path, branch):
+def clone_repo(repo_url, dest_path, branch, shallow=True):
```

Comment:
So yeah, there's one caller of `clone_repo` - it's the one that needs the changes here.

I can't imagine why a shallow clone would disturb anything else teuthology does. It's eternally possible that I've missed something, though :)